src/utils/commonGraphql.ts

```diff
@@ -0,0 +1,260 @@
+    $headerImage: String
+    $description: String
+    $repoName: String
+    $boardUrl: String
+    $skillsIds: [ID!]
+    $championsIds: [ID!]
+    $status: ProjectStatus
+  ) {
+    updateProject(
+      id: $id
+      name: $name
+      headerImage: $headerImage
+      description: $description
+      repoName: $repoName
+      boardUrl: $boardUrl
+      skillsIds: $skillsIds
+      championsIds: $championsIds
+      status: $status
+    ) {
+      ...ProjectCommon
+    }
+  }
+  ${PROJECT_COMMON_FRAGMENT}
+`;
+
+function formatProjectForUpdate(updatedProject: ProjectCommonFragment) {
+  return {
+    ...updatedProject,
+    skillsIds:
+      updatedProject.skills && updatedProject.skills.map(skill => skill.id),
+    championsIds:
+      updatedProject.champions &&
+      updatedProject.champions.map(champion => champion.id)
+  };
+}
+
+export function useUpdateProjectCommon() {
+  const updateProject = useMutation<UpdateProjectCommonMutation>(
+    UPDATE_PROJECT_COMMON,
+    {
+      // TODO: Verify these refetch queries are correct
+      // or possibly look into some way that we don't have to worry about refetching
+      refetchQueries: ['editableUsersList']
+    }
+  );
+
+  return (project: ProjectCommonFragment) =>
+    updateProject({
+      variables: formatProjectForUpdate(project)
+    });
+}
+
+export const CREATE_PROJECT_COMMON = gql`
+  mutation CreateProjectCommon(
+    $name: String!
+    $headerImage: String
+    $description: String
+    $repoName: String!
+    $boardUrl: String
+    $skillsIds: [ID!]
+    $championsIds: [ID!]
+    $status: ProjectStatus
+  ) {
+    createProject(
+      name: $name
+      headerImage: $headerImage
+      description: $description
+      repoName: $repoName
+      boardUrl: $boardUrl
+      skillsIds: $skillsIds
+      championsIds: $championsIds
+      status: $status
+    ) {
+      ...ProjectCommon
+    }
+  }
+  ${PROJECT_COMMON_FRAGMENT}
+`;
+
+function formatNewProjectForCreate(newProject: ProjectCommonFragment) {
+  if (!newProject.repoName) {
+    throw new Error('Projects require a specified repo name');
+  }
+  const { id, skills, champions, ...newProjectSansUnusedFields } = newProject;
+  return {
+    ...newProjectSansUnusedFields,
+    skillsIds: newProject.skills
+      ? newProject.skills.map(skill => skill.id)
+      : [],
+    championsIds: newProject.champions
+      ? newProject.champions.map(champion => champion.id)
+      : [],
+    repoName: newProject.repoName // Typescript was complaining without this
+  };
+}
+
+export function useCreateProjectCommon() {
+  const createProject = useMutation<CreateProjectCommonMutation>(
+    CREATE_PROJECT_COMMON,
+    {
+      // TODO: These didn't exist before, check if they are actually needed
+      refetchQueries: ['projectCards', 'projectsDrawer', 'editableUsersList']
+    }
+  );
+
+  return (newProject: ProjectCommonFragment) =>
+    createProject({
+      variables: formatNewProjectForCreate(newProject)
+    });
+}
```

Comment:
Cool, one concern I had was that some queries might be reused between components, and that eliminating the old pattern would discourage DRY code. This file addresses that 👍

Reply:
Yeah there are different tradeoffs. In Relay.js for example, I believe the recommendation is to define every graphql query (and possibly mutations too) uniquely for the specific component that uses it. I think this setup is better for our use case currently though